### PR TITLE
Simplify useFetch usage

### PR DESCRIPTION
**Pull request by chrichri17**
As discussed, we can simplify usefetch. I believe we don't need to use a store right now.

If that approach works for you, then I'll let you remove the store.

### CODE REVIEW

**Comment by chrichri17** [comment]
Current dependencies on/for this PR:
* main
  * **PR #73** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/73" target="_blank">Graphite</a> 
    * **PR #74** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/74" target="_blank">Graphite</a> 
      * **PR #77** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/77" target="_blank">Graphite</a> 
        * **PR #78** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/78" target="_blank">Graphite</a>  👈
          * **PR #79** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/79" target="_blank">Graphite</a> 
      * **PR #75** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/75" target="_blank">Graphite</a> 
        * **PR #76** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/76" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/YacineSteeve/quizoot/78?utm_source=stack-comment).

File: frontend/src/views/Quiz.vue [hunk 1]
```diff
@@ -14,85 +11,49 @@ import Loader from '@/components/Loader.vue';
...
+    return questions.value[currentQuestion.value];
 }
 
 function startQuiz() {
-    currentQuestionItem.value = getFirstQuestionItem();
-    currentQuestionRank.value = 1;
+    currentQuestion.value = 0;
 }
 
 function quitQuiz() {
```
Comment (chrichri17) [inline frontend/src/views/Quiz.vue:34]: this does nothing I think. We might want to use the router and go back to `/quizzes`

File: interfaces/quizoot.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
 export declare namespace Quizoot {
+	/**
+	 * Id of a Quizoot object
+	 *
+	 * @pattern [a-z0-9]{24}
+	 */
+	type ObjectId = string;
```
Comment (YacineSteeve) [inline interfaces/quizoot.ts:7]: Good idea

File: frontend/src/views/Home.vue [hunk 3]
```diff
@@ -1,30 +1,12 @@
...
-                log(response.error.value);
-            }
-
-            store.commit(MutationTypes.UPDATE_QUIZZES, response.data.value);
-        })
-        .catch((error) => log(error));
-}
-
-const quizzes: ComputedRef<Quiz[] | null> = computed(() => store.state.quizzes);
+const { data: quizzes } = useFetch<Quiz[]>('/api/quizzes');
```
Comment (YacineSteeve) [inline frontend/src/views/Home.vue:9]: Oh, and it works fine like that ?! Interesting

File: frontend/src/views/Quiz.vue (hunk 1 above)
Comment (YacineSteeve) [reply frontend/src/views/Quiz.vue:34]: What I meant by `quitQuiz` is actually "going back to the quiz preview".<br>
Maybe it should be renamed `quitQuestions`

And this doesn't need the router, right ?

File: frontend/src/views/Quiz.vue [hunk 4]
```diff
@@ -14,85 +11,49 @@ import Loader from '@/components/Loader.vue';
...
 <template>
-    <FetchError v-if="errorOccurred" />
-    <Loader v-else-if="isFetchingQuiz" />
+    <FetchError v-if="error" />
+    <Loader v-else-if="isFetching" />
     <div v-else class="quiz-container">
         <h1 class="quiz-title">
             <span
-                v-if="currentQuestionItem != null"
+                v-if="currentQuestion != null"
```
Comment (YacineSteeve) [inline frontend/src/views/Quiz.vue]: `v-if="currentQuestion != -1"` instead, right ?

**Review by YacineSteeve — APPROVED** [review]
I'm completely okay with this approach.

For the store, I suggest we keep the folder for now, even if it's not used anywhere.
I don't really want to re-setup it later, ha ha.
At the end, if it appears completely useless, we'll then remove it.

File: frontend/src/views/Home.vue (hunk 3 above)
Comment (chrichri17) [reply frontend/src/views/Home.vue:9]: Yup. Give it a try and let me know

File: frontend/src/views/Quiz.vue (hunk 4 above)
Comment (chrichri17) [reply frontend/src/views/Quiz.vue]: Yes. Thks for spotting that

File: frontend/src/views/Quiz.vue (hunk 1 above)
Comment (chrichri17) [reply frontend/src/views/Quiz.vue:34]: Okay.

**Pull request merged**